### PR TITLE
Allow char literal convertion into unsigned

### PR DESCRIPTION
**Pull request by ssilnicki-dev**
required to compile stm32mp1xxx dtb

### CODE REVIEW

**Comment by jlduran** [comment]
It looks fine, but let's wait for a review.
If accepted:
Close #81

File: input_buffer.cc
Comment (davidchisnall) [inline input_buffer.cc:343]: This currently won't work with escape sequences.  It would be nice to check if `outInt` is `'\'` and, if so, switch on the next character to handle `'\n'`, `'\r'`, `'\r'`, and `'\\'`, `'\''` and raise an error otherwise.

I believe the current code will error in the caller in this case.

File: input_buffer.cc
Comment (davidchisnall) [inline input_buffer.cc:885]: I think that most of this logic would make more sense in `consume_char_literal`, but I don't have a very strong opinion here.

**Comment by davidchisnall** [comment]
Thanks.  I am happy to merge this, but I'd prefer if we handled common escape characters in character literals first.

File: input_buffer.cc
Comment (ssilnicki-dev) [reply input_buffer.cc:343]: I didn't pay attention to such cases. I'll update

**Review by davidchisnall — COMMENTED** [review]
Can you also add a test?  Tests for things like this are simple: just a .dts file using the feature and a .dts.expected file that contains the expected output (you can generate this with `-I dts -O dts` and visually check the output).

File: input_buffer.cc
Comment (ssilnicki-dev) [reply input_buffer.cc:363]: ok.

**Review by ssilnicki-dev — COMMENTED** [review]
I think, '\0' may be supported right now, before octal or hex escapes implemented.

**Review by ssilnicki-dev — COMMENTED** [review]
I assume, all commits will be squashed when merging this PR.

File: input_buffer.cc
Comment (davidchisnall) [inline input_buffer.cc:373]: Minor style nit, but it’s good to put default first so that nothing accidentally falls through into it.

**Review by davidchisnall — APPROVED** [review]
Thanks, looks good.

I can squash and merge, unless you want to tidy up the history first.

File: input_buffer.cc
Comment (ssilnicki-dev) [reply input_buffer.cc:373]: ok. minute

**Comment by ssilnicki-dev** [comment]
please squash & merge. I prefer to keep history

**Comment by davidchisnall** [comment]
Thanks!

**Comment by jlduran** [comment]
Thank you!
I'll try downstreaming it as soon as possible:
https://reviews.freebsd.org/D42438

**Comment by ssilnicki-dev** [comment]
@jlduran Dear Jose, can you push the patch further from fabricator into current? It seems that port's version is now patched, but it is not convenient to work with two versions of dtc installed.

**Comment by jlduran** [comment]
> @jlduran Dear Jose, can you push the patch further from fabricator into current? It seems that port's version is now patched, but it is not convenient to work with two versions of dtc installed.

Done!

File: input_buffer.cc
Comment (jrtc27) [inline input_buffer.cc:346]: if (

File: input_buffer.cc
Comment (jrtc27) [inline input_buffer.cc:350]: if (

File: input_buffer.cc
Comment (jrtc27) [inline input_buffer.cc:920]: if (

Comment (ssilnicki-dev) [reply input_buffer.cc:920]: @jlduran Do I need to engage?

Comment (jlduran) [reply input_buffer.cc:920]: I'll submit a PR here addressing @jrtc27 style(9) changes.

(I was trying to add a super-linter to the project, to avoid this in the future, but it'll take a while).

Comment (jlduran) [reply input_buffer.cc:920]: Done in #85

Will soon fix downstream (FreeBSD) as well. Thank you!

**Pull request merged**
